allennlp/training/trainer.py

```diff
@@ -532,7 +538,7 @@ def _save_checkpoint(self, epoch: Union[int, str]) -> None:
                 training_states=training_states,
                 is_best_so_far=self._metric_tracker.is_best_so_far())
 
-    def _restore_checkpoint(self) -> int:
+    def _restore_checkpoint(self) -> Tuple[int, int, Dict[str, float]]:
```

Comment:
for future reference, I will probably never approve a PR that contains a return type like this :innocent: 

in this particular case, I don't see why this method needs to return the best_epoch or the best_epoch_metrics when you can just get them from the metric_tracker

Reply:
Well, but that's exactly what I had done earlier, before you suggested to move the logic inside `_restore_checkpoint` (because the logic belongs there). My interpretation was that the point of `_restore_checkpoint` is to do all the things required to restore the training and model state (wrapping `self._checkpointer` and `self._metric_tracker`) so logic to get `best_epoch` and `best_epoch_metrics ` belongs inside and not outside of it.

Reply:
the *logic* belongs there, but the function doesn't need to return them explicitly, because after restore_checkpoint they should have been restored as part of the metric tracker?

Reply:
.. may be I misinterpreted. So let me know if so.

Reply:
Cool, no problem. I will revert it back and take care in future 👍.

The only *movable logic* at the point you commented was using `self._metric_tracker.best_epoch` and `self._metric_tracker.best_epoch_metrics`. So, I probably misinterpreted that to be moved up.